packages/salesforcedx-vscode-core/src/commands/forceSourceRetrieveCmp.ts

```diff
@@ -39,20 +39,33 @@ import {
 export class ForceSourceRetrieveExecutor extends SfdxCommandletExecutor<
   string
 > {
-  private metadataArg: string;
-  constructor(metadataArg: string) {
+  private typeName: string;
+  private componentName: string;
+
+  constructor(typeName: string, componentName: string) {
     super();
-    this.metadataArg = metadataArg;
+    this.typeName = typeName;
+    this.componentName = componentName;
   }
 
   public build(): Command {
     return new SfdxCommandBuilder()
       .withDescription(nls.localize('force_source_retrieve_text'))
       .withArg('force:source:retrieve')
-      .withFlag('-m', this.metadataArg)
+      .withFlag('-m', `${this.typeName}:${this.componentName}`)
       .withLogName('force_source_retrieve')
       .build();
   }
+
+  protected getTelemetryData(): TelemetryData {
+    const retrievedTypes: any = {};
+    retrievedTypes[this.typeName] = 1;
+    return {
+      properties: {
+        'org-browser/retrievedTypes': JSON.stringify(retrievedTypes)
```

Comment:
1. I'm trying to future proof this so when we add the ability to retrieve multiple components, we know all the types and the number of components for each that was retrieved. This will require some query building kung fu to parse the json.

2. I'm thinking about adding a prefix (in this case **org-browser**) to properties so we have a better idea of what they are used for in dashboards, since we're starting to have a decent amount of them. Any strong feelings on this?

Reply:
Looks good to me, we should keep the prefix.